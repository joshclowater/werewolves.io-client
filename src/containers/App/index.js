import React, { Fragment, Suspense, lazy } from 'react';
import { HashRouter, Route } from 'react-router-dom';

import CenteredScreen from 'src/components/Centered/CenteredScreen';
import LoadingMessage from 'src/components/LoadingMessage';
import LandingPage from 'src/containers/LandingPage';

const Host = lazy(() => import('src/containers/Host'));
const Player = lazy(() => import('src/containers/Player'));

const fallback = (
  <CenteredScreen>
    <LoadingMessage message="Loading game" />
  </CenteredScreen>
);

export default function App() {
  return (
    <HashRouter>
      <Suspense fallback={fallback}>
        <Fragment>
          <Route exact path="/" component={LandingPage} />
          <Route path="/host" component={Host} />
          <Route path="/player" component={Player} />
        </Fragment>
      </Suspense>
    </HashRouter>
  );
}
