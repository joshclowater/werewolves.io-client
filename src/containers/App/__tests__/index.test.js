import React from 'react';
import { shallow } from 'enzyme';
import App from '../';

it('renders correctly', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
