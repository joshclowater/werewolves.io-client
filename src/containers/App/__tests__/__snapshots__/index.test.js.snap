// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Host)",
        "type": [Function],
      }
    }
    path="/host"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Player)",
        "type": [Function],
      }
    }
    path="/player"
  />
</BrowserRouter>
`;
